src/os_win/os_thread.c

```diff
@@ -33,10 +33,15 @@ __wt_thread_join(WT_SESSION_IMPL *session, wt_thread_t tid)
 {
 	WT_DECL_RET;
 
-	if ((ret = WaitForSingleObject(tid, INFINITE)) == WAIT_OBJECT_0)
-		return (0);
+	if ((ret = WaitForSingleObject(tid, INFINITE)) != WAIT_OBJECT_0)
+		/*
+		 * If we fail to wait, we will leak handles so do not continue
+		 */
+		WT_PANIC_RET(session, WT_PANIC, "Wait for thread join failed");
+
+	CloseHandle(tid);
 
-	WT_RET_MSG(session, ret, "WaitForSingleObject");
+	return (0);
```

Comment:
@keithbostic 
1. Good suggestion. I was unsure what to pass to WT_PANIC in this case for that argument.
2. I cleaned up CloseHandle in this function, and everywhere else.